python/grass/temporal/abstract_map_dataset.py

```diff
@@ -165,33 +166,52 @@ def get_map_id(self):
         return self.base.get_map_id()
 
     @staticmethod
-    def build_id(name, mapset, layer=None):
+    def build_id(name, mapset, layer=None, check_mapset_element=None):
         """Convenient method to build the unique identifier
 
         Existing layer and mapset definitions in the name
         string will be reused
 
+        If check_mapset_element is given, but no mapset, the function will
+        try to get the correct mapset by looking for an element of the
+        given element type with the given name. If the combination is
+        not found on the current search path, it will fail and throw an error.
+
```

Comment:
Perhaps add that check is not executed if mapset is given. Given the current parameter name `check`, this behavior seems unexpected to me. Would it make sense to check it in any case?

Reply:
Or, change the parameter name to `element`.